opennav_docking_core/include/opennav_docking_core/charging_dock.hpp

```diff
@@ -81,7 +81,7 @@ class ChargingDock
    * @param pose The initial estimate of the dock pose.
    * @param frame The frame of the initial estimate.
    */
-  virtual bool getRefinedPose(geometry_msgs::msg::PoseStamped & pose) = 0;
+  virtual bool getRefinedPose(geometry_msgs::msg::PoseStamped & pose, std::string id) = 0;
```

Comment:
This biggest issue with a backport to humble is this - if others have made their own plugins already, this would break them with an update as a hard API breaking change.

I think for backporting, it would need to be `std::string id = std::string()` so that if its not implemented, that's OK. But I'd test that with the simple charging dock plugin to make sure that all renders OK if you don't include the ID in the override definition (which I think will cause a problem).